### PR TITLE
Made all network connections proxy-compatible

### LINKED ISSUE #243: proxy is not used everywhere
In some places the installer still use "file_get_contents", which does not work behind a proxy.
e.g.
- DownloadCommand::createGitIgnore
- SelfUpdateCommand...
- ...

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #243.

### CODE REVIEW

File: src/Symfony/Installer/SelfUpdateCommand.php [hunk 1]
```diff
@@ -98,16 +97,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
                 ->cleanUp()
             ;
         } catch (IOException $e) {
+            if ($output->isVeryVerbose()) {
+                echo $e->getMessage();
```
Comment (Pierstoval) [inline src/Symfony/Installer/SelfUpdateCommand.php]: You could just add the `$e` var as third argument of the `RuntimeException` thrown right after it

Comment (stof) [inline src/Symfony/Installer/SelfUpdateCommand.php]: this should use the output btw, not a straight ``echo``

File: src/Symfony/Installer/SelfUpdateCommand.php [hunk 2]
```diff
@@ -85,8 +82,10 @@ protected function initialize(InputInterface $input, OutputInterface $output)
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        if ($this->installerIsUpdated()) {
-            return;
+        if ($this->isInstallerUpdated()) {
+            $this->output->writeln(sprintf('// Symfony Installer is <info>already updated</info> to the latest version (%s).', $this->latestInstallerVersion));
```
Comment (stof) [inline src/Symfony/Installer/SelfUpdateCommand.php]: missing return after that, to avoid downloading it

**Comment by stof** [comment]
Why is the refactoring of the installer version check shipped in the same PR (without being described in the PR) ?

**Comment by javiereguiluz** [comment]
@stof I agree that's a bad practice and I'm sorry about it.

**Comment by stof** [comment]
@javiereguiluz note that this hidden change is even the one preventing to merge this PR, as it is the buggy part...

**Pull request closed without merging**